File: src/main/java/com/camackenzie/exvi/core/ExerciseForceType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core;

/**
 *
 * @author callum
 */
public enum ExerciseForceType {
    PUSH,
    PULL,
    DYNAMIC_STRETCH,
    STATIC_STRETCH,
    COMPRESSION,
    ISOMETRIC,
    STATIC,
    HINGE;

    public static ExerciseForceType fromString(String s) {
        return EnumUtils.enumFromString(ExerciseForceType.class, s);
    }

    @Override
    public String toString() {
        return EnumUtils.formatName(super.toString());
    }
}

File: exvi-core/src/commonMain/kotlin/com/camackenzie/exvi/core/model/ExerciseForceType.kt
/*
 * Copyright (c) Callum Mackenzie 2022.
 */

package com.camackenzie.exvi.core.model

import com.camackenzie.exvi.core.util.EnumUtils
import com.camackenzie.exvi.core.util.EnumUtils.formatName

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
@Suppress("unused")
enum class ExerciseForceType {
    Push,
    Pull,
    DynamicStretching,
    StaticStretching,
    Compression,
    Isometric,
    Static,
    Hinge,
    Other;

    override fun toString(): String = formatName(super.toString())

    companion object {
        @kotlin.jvm.JvmStatic
        fun fromString(s: String): ExerciseForceType? = EnumUtils.enumFromString<ExerciseForceType>(s)
    }
}
